src/OrchardCore.Modules/Orchard.Users/Services/UserStore.cs

```diff
@@ -322,19 +346,41 @@ public Task<IList<string>> GetRolesAsync(User user, CancellationToken cancellati
             return Task.FromResult<IList<string>>(user.RoleNames);
         }
 
-        public Task<bool> IsInRoleAsync(User user, string roleName, CancellationToken cancellationToken)
+        public async Task<bool> IsInRoleAsync(User user, string normalizedRoleName, CancellationToken cancellationToken = default(CancellationToken))
         {
             if (user == null)
             {
                 throw new ArgumentNullException(nameof(user));
             }
 
-            return Task.FromResult(user.RoleNames.Contains(roleName));
+            if (string.IsNullOrWhiteSpace(normalizedRoleName))
+            {
+                throw new ArgumentException(T["Value cannot be null or empty."], nameof(normalizedRoleName));
+            }
+
+            var roleNames = await _roleProvider.GetRoleNamesAsync();
```

Comment:
And maybe normalize both the roles that are in the user object and the string passed in the method, this way we can also pass not normalized roles.